python/dgl/frame.py

```diff
@@ -548,7 +559,14 @@ def record_stream(self, stream):
         """
         if F.get_preferred_backend() != "pytorch":
             raise DGLError("record_stream only supports the PyTorch backend.")
-        self.data.record_stream(stream)
+        # Only record streams for materialized tensors otherwise just record for the index
```

Comment:
Could there be an issue if `record_stream` gets called followed by a call to
```
@data.setter
def data(self, val):
```
Or does it make more sense that when assigning data or an index to a column, the stream associated with that column becomes the from which the data or index is associated it?

Reply:
That sounds good to me